src/SkynetServer/Network/Client.cs

```diff
@@ -0,0 +1,66 @@
+﻿using SkynetServer.Entities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading.Tasks;
+using VSL;
+
+namespace SkynetServer.Network
+{
+    internal partial class Client : IVSLCallback, IPacketHandler
+    {
+        private VSLServer socket;
+        private Account account;
+        private Session session;
+
+        public async Task SendPacket(Packet packet)
+        {
+            using (var buffer = PacketBuffer.CreateDynamic())
+            {
+                packet.WritePacket(buffer);
+                bool success = await socket.SendPacketAsync(packet.Id, buffer.ToArray());
+                if (!success)
+                    Console.WriteLine($"Failed to send packet {packet.Id}");
```

Comment:
shouldn we close the connection here?

Reply:
VSL will automatically call `IVSLCallback.OnConnectionClosed()` when it fails to send a message. The Console output is only for debugging purposes like everywhere else as we don't have a logging API yet.